app/src/main/kotlin/com/wire/android/ui/home/gallery/MediaGalleryViewModel.kt

```diff
@@ -107,15 +108,19 @@ class MediaGalleryViewModel @Inject constructor(
 
     fun deleteCurrentImageMessage() {
         viewModelScope.launch {
+            navigationManager.navigateBack
             navigationManager.navigate(
```

Comment:
I was thinking actually of adding arguments to `navigateBack`, because right now when using `navigate` with `BackStackMode.CLEAR_CURRENT`, `destination` isn't even used because it's only popping backstack, so for me using it this way is a little bit confusing as `navigate` should go forward, and `navigateBack` backwards.
What I would do is just remove this `CLEAR_CURRENT` mode completely, and instead modify `NavigationManager` this way:
```
var navigateBack = MutableSharedFlow<List<Pair<String, Any>>>()
...
suspend fun navigateBack(previousBackStackPassedArgs: List<Pair<String, Any>> = listOf()) {
    navigateBack.emit(previousBackStackPassedArgs)
}
```
and then in `WireActivity` just:
```
navigationManager.navigateBack
      .onEach {
             keyboardController?.hide()
             popWithArguments(navController, it)
      }
      .launchIn(scope)
```
Then here in `deleteCurrentImageMessage()` you could just use 
```
navigationManager.navigateBack(
    listOf(
        EXTRA_MESSAGE_TO_DELETE_ID to imageAssetId.messageId,
        EXTRA_MESSAGE_TO_DELETE_IS_SELF to imageAssetId.isSelfAsset
     )
)
```
instead of `navitagionManager.navigate(...)`

What do you think about that?

BTW we can use `Map<String, Any>` instead of `List<Pair<String, Any>>` 😅

Reply:
Awesome suggestion, it will save ourselves from having to pass that extra `MessageDeletion` object on the `Conversation` path 🙏